tensorflow_io/core/kernels/serialization_kernels.cc

```diff
@@ -58,141 +57,94 @@ class DecodeJSONOp : public OpKernel {
-                value_tensor->flat<tstring>()(j) = (*entry)[j].GetString();
-              }
-              break;
-            default:
-              OP_REQUIRES(context, false,
-                          errors::InvalidArgument(
-                              "data type not supported: ",
-                              DataTypeString(value_tensor->dtype())));
-              break;
-          }
-        }
-
-      } else {
-        OP_REQUIRES_OK(context, context->allocate_output(i, TensorShape({1}),
-                                                         &value_tensor));
-        switch (value_tensor->dtype()) {
-          case DT_INT32:
-            value_tensor->flat<int32>()(0) = entry->GetInt();
-            break;
-          case DT_INT64:
-            value_tensor->flat<int64>()(0) = entry->GetInt64();
-            break;
-          case DT_FLOAT:
-            value_tensor->flat<float>()(0) = entry->GetDouble();
-            break;
-          case DT_DOUBLE:
-            value_tensor->flat<double>()(0) = entry->GetDouble();
-            break;
-          case DT_STRING:
-            value_tensor->flat<tstring>()(0) = entry->GetString();
-            break;
-          default:
-            OP_REQUIRES(
-                context, false,
-                errors::InvalidArgument("data type not supported: ",
-                                        DataTypeString(value_tensor->dtype())));
-            break;
-        }
+      std::vector<int64> tensor_shape_vector;
+      getTensorShape(entry, tensor_shape_vector);
+      OP_REQUIRES_OK(
+          context, context->allocate_output(i, TensorShape(tensor_shape_vector),
+                                            &value_tensor));
+      int64 flat_index;
+      flat_index = 0;
+      switch (value_tensor->dtype()) {
+        case DT_INT32:
+          writeToTensor(entry, value_tensor, flat_index, writeInt32);
+          break;
+        case DT_INT64:
+          writeToTensor(entry, value_tensor, flat_index, writeInt64);
+          break;
+        case DT_FLOAT:
+          writeToTensor(entry, value_tensor, flat_index, writeFloat);
+          break;
+        case DT_DOUBLE:
+          writeToTensor(entry, value_tensor, flat_index, writeDouble);
+          break;
+        case DT_STRING:
+          writeToTensor(entry, value_tensor, flat_index, writeString);
+          break;
+        default:
+          OP_REQUIRES(
+              context, false,
+              errors::InvalidArgument("data type not supported: ",
+                                      DataTypeString(value_tensor->dtype())));
+          break;
       }
     }
   }
 
  private:
   mutable mutex mu_;
   Env* env_ TF_GUARDED_BY(mu_);
+
+  // Tensor Shape
+
+  static void getTensorShape(rapidjson::Value* entry,
+                             std::vector<int64>& tensor_shape_vector) {
+    if (entry->IsArray()) {
+      tensor_shape_vector.push_back(entry->Size());
+      getTensorShape(&(*entry)[0], tensor_shape_vector);
+    }
+  }
```

Comment:
@djl11 if the entry is not an array, shouldn't we assign it a constant shape?

Reply:
Yes I missed this. So [originally](https://github.com/tensorflow/io/blob/master/tensorflow_io/core/kernels/serialization_kernels.cc#L164) this was just set as 1. I will make the change now.